admin/tabs/roles/RolesModel.ts

```diff
@@ -0,0 +1,280 @@
+                filterable: true
+            },
+            columns: [
+                {field: {name: 'name', type: 'string'}},
+                {field: {name: 'category', type: 'string'}},
+                {field: {name: 'lastUpdated', type: 'date'}, ...Col.dateTime},
+                {field: {name: 'lastUpdatedBy', type: 'string'}},
+                {field: {name: 'notes', type: 'string'}, filterable: false, flex: 1}
+            ],
+            contextMenu: [...this.ACTIONS, '-', ...GridModel.defaultContextMenu]
+        });
+    }
+
+    private createRecordActions(): RecordActionSpec[] {
+        return [
+            {
+                text: 'Add',
+                icon: Icon.add(),
+                intent: 'success',
+                actionFn: () => this.createAsync()
+            },
+            {
+                text: 'Edit',
+                icon: Icon.edit(),
+                intent: 'primary',
+                actionFn: ({record}) => this.editAsync(record.data as HoistRole),
+                recordsRequired: true
+            },
+            {
+                text: 'Clone',
+                icon: Icon.copy(),
+                actionFn: ({record}) => this.createAsync(record.data as HoistRole),
+                recordsRequired: true
+            },
+            {
+                text: 'Delete',
+                icon: Icon.delete(),
+                intent: 'danger',
+                actionFn: ({record}) =>
+                    this.deleteAsync(record.data as HoistRole)
+                        .catchDefault()
+                        .linkTo(this.loadModel),
+                recordsRequired: true
+            }
+        ];
+    }
+
+    private createFilterChooserModel(): FilterChooserModel {
+        return new FilterChooserModel({
+            bind: this.gridModel.store,
+            fieldSpecs: [
+                'name',
+                'category',
+                'users',
+                'directoryGroups',
+                'roles',
+                'inheritedRoleNames',
+                'effectiveUserNames',
+                'effectiveDirectoryGroupNames',
+                'effectiveRoleNames',
+                'lastUpdatedBy',
+                {
+                    field: 'lastUpdated',
+                    example: 'YYYY-MM-DD',
+                    valueParser: (v, op) => {
+                        let ret = moment(v, ['YYYY-MM-DD', 'YYYYMMDD'], true);
+                        if (!ret.isValid()) return null;
+
+                        // Note special handling for '>' & '<=' queries.
+                        if (['>', '<='].includes(op)) {
+                            ret = moment(ret).endOf('day');
+                        }
+
+                        return ret.toDate();
+                    },
+                    valueRenderer: v => fmtDate(v),
+                    ops: ['>', '>=', '<', '<=']
+                }
+            ],
+            persistWith: {...RolesModel.PERSIST_WITH, path: 'mainFilterChooser'}
```

Comment:
I would not persist filters - I consider it an anti-pattern in an admin UI like this.  You always want to come back to it fresh seeing all the data.